cmd/cloud-controller-manager/gkenetworkparamsetcontroller.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	cloudprovider "k8s.io/cloud-provider"
+	networkclientset "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned"
+	networkinformers "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions"
+	gkenetworkparamsetcontroller "k8s.io/cloud-provider-gcp/pkg/controller/gkenetworkparamset"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	"k8s.io/cloud-provider/app"
+	cloudcontrollerconfig "k8s.io/cloud-provider/app/config"
+	genericcontrollermanager "k8s.io/controller-manager/app"
+	"k8s.io/controller-manager/controller"
+	kubecontroller "k8s.io/kubernetes/pkg/controller"
+)
+
+const jsonContentType = "application/json"
+
+func startGkeNetworkParamSetControllerWrapper(initCtx app.ControllerInitContext, config *cloudcontrollerconfig.CompletedConfig, c cloudprovider.Interface) app.InitFunc {
+	return func(ctx context.Context, controllerCtx genericcontrollermanager.ControllerContext) (controller.Interface, bool, error) {
+		return startGkeNetworkParamsController(config, controllerCtx, c)
+	}
+}
+
+func startGkeNetworkParamsController(ccmConfig *cloudcontrollerconfig.CompletedConfig, controllerCtx genericcontrollermanager.ControllerContext, cloud cloudprovider.Interface) (controller.Interface, bool, error) {
+
+	gceCloud, ok := cloud.(*gce.Cloud)
+	if !ok {
+		err := fmt.Errorf("GkeNetworkParamsController does not support %v provider", cloud.ProviderName())
+		return nil, false, err
+	}
+
+	kubeConfig := ccmConfig.Complete().Kubeconfig
+	kubeConfig.ContentType = jsonContentType //required to serialize GKENetworkParamSet to json
+
+	networkClient, err := networkclientset.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, false, err
+	}
+
+	//no resync, we dont want to automatically update objects if their state changes in gcp
+	networkInformerFactory := networkinformers.NewSharedInformerFactory(networkClient, kubecontroller.NoResyncPeriodFunc())
```

Comment:
You don't need a shared informer since you are only watching only one resource, there should be one generated Informer builder for this Resource, something as `NewGKENetworkParamSetsInformer`